### DISCUSSION TITLE
Trigger rule on repeated state, not only on state changes

### DISCUSSION THREAD

**Discussion by elbman** [arendst/Tasmota#12467] [Ideas]
I'm using Wemos D1 Mini configured with 8 Relay (virtually) only for HUE Bridge Emulation to interact with Echo/Alexa.

configured rules to execute commands like websend (i.e. to switch TV ON/OFF over IP)

`Rule1
ON Power1#State=0 DO WebSend [192.168.2.4] /config/xmlapi/statechange.cgi?ise_id=12110&new_value=0 ENDON 
ON Power1#State=1 DO WebSend [192.168.2.4] /config/xmlapi/statechange.cgi?ise_id=12110&new_value=1 ENDON
...
`

This Solution works very well, 

BUT:  

After you switched ON the TV with the remote control,  you can't say "Alexa, switch OFF TV" 

because Power1#State is still OFF, Power1#State does not change so nothing happens via Rule

Is there a better trigger to determine the on / off command comming from Alexa via HUE Bridge Emulation?

Other ideas?

**Comment by Jason2866** [comment]
I have solved such a situation with my HomeAutomation system which is connected to the Alexa API to sync states. I get the state with a Tasmota IR receiver which captures the remote commands.

**Reply by elbman** [reply]
Hey, nice solution! I can control my LG TV directly over IP, and the Remote Control is mostly working on Bluetooth.

(To get the state it would be possible to ping the device, but that is not the best solution)


There must be a trigger instead of Power1#State that determines the on / off command comming from Alexa

Alexa has to work for me like a Button, when pressed (ON oder OFF) a command should be executed

Maybe there is a special MODE that allows this behaviour

**Reply by Jason2866** [reply]
Alexa API is described https://developer.amazon.com/en-US/docs/alexa/device-apis/alexa-powercontroller.html
There is no toggle

**Reply by elbman** [reply]
Thank you. But i need no toogle Funktion. 

I just want to give an „Alexa Switch ON“ command several times … 
I think Tasmota regognises the ON from Alexa, but nothing to do because PowerState is already ON

Therefore the question how to detect / trigger this event 🤷‍♂️

I used HABridge bevore with a raspberry pi to emulate a HUE Bridge an send a http command to my TV. Every Time you say „Switch ON“ the command was executed.

**Comment by elbman** [comment]
Tasmota Console shows me, the Alexa command to switch ON, again and again...

Rule is not executed because Power#State is the same, only time changes

09:10:55.153 RSL: RESULT = {"POWER1":"ON"}
09:10:55.155 RSL: POWER1 = ON
09:11:48.947 RSL: RESULT = {"POWER1":"ON"}
09:11:48.949 RSL: POWER1 = ON

How can i catch this event for my rule?

Is there realy no solution?

**Reply by barbudor** [reply]
I can confirm that even if you do not enable the "once" mode of the rule, it will trigger only on state changes, not on repeated states
Just verified that basically on a nodemcu and issuing console power commands

**Comment by elbman** [comment]
I am a bit surprised that nobody has had this problem so far

**would it be possible to implement something like a new SetOption

so that a rule also triggers on repeated states, not only on state changes ?**

that would in my eyes significantly expand the possibilities of interacting with Alexa! 

You than can switch on/off things that don't give feedback / sync their states.

(sending emails, clear something like cache, get your own sensor information with Text2Speech, etc.)

**Reply by s-hadinger** [reply]
I will have a look.

**Reply by sfromis** [reply]
One way of getting what you ask for is to use `SetOption59` to add state reporting MQTT messages when power state is updated. Then you can parse power state, changed or not, using a rule like this:
`on tele-Power do var1 %value% endon`
This will also trigger at the telemetry interval.

**Reply by elbman** [reply]
hm, i tried this out and can trigger (toggle), but

now rule is also executed every 300 seconds and i can't differ between ON and OFF

(i.e. important for my TV)

**Reply by sfromis** [reply]
If you include a comparison in the rule, you can make it depend on the value, like `on tele-Power=ON do ....`.
It is possible to disable messages at telemetry period using the TelePeriod command, but of course than you'd also lose other info.

Still, having an unchanged power state trigger a rule would likely also trigger when using the Power command to just query power state.

**Reply by elbman** [reply]
it works nearly

TelePeriod 0 

Rule1
ON tele-Power1=OFF DO WebSend [192.168.2.4] /value=0 ENDON ON tele-Power1=ON DO WebSend [192.168.2.4] /value=1 ENDON
ON tele-Power2=OFF DO WebSend [192.168.2.4] /value=0 ENDON ON tele-Power2=ON DO WebSend [192.168.2.4] /value=1 ENDON
ON tele-Power3=OFF DO WebSend [192.168.2.4] /value=0 ENDON ON tele-Power3=ON DO WebSend [192.168.2.4] /value=1
ENDON
ON tele-Power4=OFF DO WebSend [192.168.2.4] /value=0 ENDON ON tele-Power4=ON DO WebSend [192.168.2.4] /value=1 ENDON

now repeating one state or switching it executes all commands in dependence on their powerstates (1-4) ???

Repeat Power4 with Alexa now results in:

21:02:51.377 RSL: STATE = {"Time":"2021-06-26T21:02:51","Uptime":"0T07:39:17","UptimeSec":27557,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":21,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"OFF","POWER5":"OFF","POWER6":"OFF","POWER7":"OFF","POWER8":"OFF","Wifi":{"AP":1,"SSId":"xxx","BSSId":"xxx","Channel":11,"Mode":"11n","RSSI":100,"Signal":-36,"LinkCount":1,"Downtime":"0T00:00:04"}}
21:02:51.393 RUL: TELE-POWER1=OFF performs "WebSend [192.168.2.4] /value=0"
21:02:51.441 RSL: RESULT = {"WebSend":"Done"}
21:02:51.459 RUL: TELE-POWER2=OFF performs "WebSend [192.168.2.4] /value=0"
21:02:51.506 RSL: RESULT = {"WebSend":"Done"}
21:02:51.523 RUL: TELE-POWER3=OFF performs "WebSend [192.168.2.4] /value=0"
21:02:51.567 RSL: RESULT = {"WebSend":"Done"}
21:02:51.585 RUL: TELE-POWER4=OFF performs "WebSend [192.168.2.4] /value=0"
21:02:51.624 RSL: RESULT = {"WebSend":"Done"}
21:02:51.631 RSL: RESULT = {"POWER4":"OFF"}
21:02:51.633 RSL: POWER4 = OFF

**Reply by sfromis** [reply]
Well, you only asked about one power state ;-) And yes, the /STATE payload will contain all state, changed or not.

**Reply by elbman** [reply]
i wanted to use up to 8 virtual devices separately, (TV on/off, Mute on/off, Play/Pause etc. ...), 

triggering repeated states on each channel

Unfortunately SetOption59 and MQTT does not seem to be the solution here.

**Reply by elbman** [reply]
I hope it's ok to warm up my issue. 

@s-hadinger: do you (or maybe someone else) have an idea or already a solution for this?
